### PR TITLE
BrandAPI: fixes for Java SDK

### PR DESCRIPTION
**Pull request by sergiishamrai-okta**
Issue addressed:

Since `AccessPolicyRuleConditions` inherits `PolicyRuleConditions` and has override method `getDevice()` the return type for `getDevice()` method should has covariant return type. Therefore, `DeviceAccessPolicyRuleCondition` should inherit `DevicePolicyRuleCondition`.

### CODE REVIEW

File: dist/spec.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
       "name": "Apache-2.0",
       "url": "https://www.apache.org/licenses/LICENSE-2.0.html"
     },
-    "version": "2.9.1"
+    "version": "2.9.2"
```
Comment (arvindkrishnakumar-okta) [inline dist/spec.json:16]: should this be a major rev instead?

Comment (sergiishamrai-okta) [reply dist/spec.json:16]: This should not be a breaking change for other SDKs. So IMHO it could be minor rev update

**Review by arvindkrishnakumar-okta — APPROVED** [review]
LGTM

**Comment by sergiishamrai-okta** [comment]
@laura-rodriguez @arvindkrishnakumar-okta 
`ThemeResponse` changes reverted

**Review by oleksandrpravosudko-okta — APPROVED** [review]

**Review by VitaliiTytarenko-okta — APPROVED** [review]

**Review by serhiibuniak-okta — APPROVED** [review]
LGTM, Python SDK works

**Review by laura-rodriguez — APPROVED** [review]

**Pull request merged**